src/st/st0/30030.c

```diff
@@ -105,7 +105,364 @@ u16 func_801B0414(void) {
     return ret;
 }
 
-INCLUDE_ASM("st/st0/nonmatchings/30030", func_801B0464);
+// func_801B0414 is undeclared to PrologueScroll. This emulates the
+// undefined signature to avoid splitting the file at this time.
+#define fake_func_801B0414 ((s32(*)())func_801B0414)
+void func_801B1298(Entity*);
+
+typedef struct ProloguePrimitive {
+    s8 u0;
+    s8 v0;
+    s8 u1;
+    s8 v1;
+    s16 x;
+    s16 y;
+    u16 tpage;
+} ProloguePrimitive;
+
+extern s32 D_8003CB04[];
+extern Entity D_800762D8;
+extern ProloguePrimitive D_80181568[];
+extern u16 D_801BEDFC;
+extern u16 D_801BEE00;
+extern u16 D_801BEE04;
+extern s16 D_801BEE08;
+extern s16 D_801BEE0C;
+
+// Post-Dracula fight prologue scroll. This is responsible for scrolling the
+// prologue text and displaying the Castlevania logo.
+void PrologueScroll(void) {
+    Entity* entity;
+    ProloguePrimitive* textPrim;
+    Primitive* prim;
+    s32 primIndex;
+    s32 condition;
+    s32 i;
+
+    entity = &D_800762D8;
+
+    switch (g_GameStep) {
+    case 0: // 0x801B04A4
+        if (g_UseDisk) {
+            if (!g_IsUsingCd) {
+                g_CdStep = 1;
+                g_LoadFile = 6;
+            } else {
+                break;
+            }
+        }
+
+        D_801BEE08 = g_api.AllocPrimitives(PRIM_SPRT, 0x10);
+        prim = &g_PrimBuf[D_801BEE08];
+
+        for (i = 0; i < 2; i++) {
+            prim->x0 = (i * 0xb0) + 0x50;
+            prim->y0 = 0x15;
+            prim->u0 = i ? 0x50 : 0x20;
+
+            prim->v0 = 0;
+            prim->v1 = prim->u1 = 0xB0;
+            prim->tpage = i + 0x8D;
+            prim->clut = 0x210;
+            prim->priority = 0;
+            prim->drawMode = 8;
```

Comment:
#defines for all the drawModes https://github.com/Xeeynamo/sotn-decomp/blob/6462f2f4eb2f564ddf9e51777b5cd711983b4068/include/game.h#L81